sample/composeApp/src/commonMain/kotlin/com/mifos/passcode/sample/platformAuthentication/AuthenticationScreen.kt

```diff
@@ -0,0 +1,189 @@
+package com.mifos.passcode.sample.platformAuthentication
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Color.Companion.White
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.navigation.NavController
+import auth.deviceAuth.AuthenticationResult
+import com.mifos.passcode.LibraryLocalPlatformAuthenticationProvider
+import com.mifos.passcode.LibraryPlatformAvailableAuthenticationOption
+import com.mifos.passcode.auth.deviceAuth.PlatformAuthenticatorStatus
+import com.mifos.passcode.auth.passcode.components.MifosIcon
+import com.mifos.passcode.sample.chooseAuthOption.DialogBoxType
+import com.mifos.passcode.sample.chooseAuthOption.MessageDiaglogBox
+import com.mifos.passcode.sample.navigation.Route
+import com.mifos.passcode.sample.platformAuthentication.components.SystemAuthenticatorButton
+import com.mifos.passcode.ui.theme.blueTint
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun AuthenticationScreen(
+    authenticationScreenViewModel: AuthenticationScreenViewModel,
+    navController: NavController,
+) {
+    val verificationResult = authenticationScreenViewModel.authenticationResult.collectAsState()
+
```

Comment:
in mobile the `collectAsStateWithLifecycle()` is preffered than `collectAsState` so use that